ui/org.eclipse.pde.core/src/org/eclipse/pde/core/project/IHostDescription.java

```diff
@@ -30,14 +30,25 @@ public interface IHostDescription {
 	 *
 	 * @return symbolic name of the host
 	 */
-	public String getName();
+	String getName();
 
 	/**
 	 * Returns the version constraint of the host or <code>null</code>
 	 * if unspecified.
 	 *
 	 * @return version constraint or <code>null</code>
+	 * @since 3.19
 	 */
-	public VersionRange getVersionRange();
+	VersionRange getVersion();
+	// TODO: alternatively name it getHostVersionRange() ?
```

Comment:
Since only the return type changes from `org.eclipse.osgi.service.resolver.VersionRange` to `org.osgi.framework.VersionRange` to we have to find a new name for the method.
The OSGi spec only talks about the _version_ of a requirement (e.g. host, package or bundle) although technically it can be a version range. With that in mind I found naming the new method just `getVersion()` a good choice, especially since there is no need to reflect the exact return type in a method name. The method name should reflect the semantic of a method.
Alternatively we could name it for example `getHostVersionRange()`. But not only would this 'duplicate' the return type information from the method signature but also the information of the containing class.

The same applies for `IPackageImportDescription.getVersion()` and `IRequiredBundleDescription.getVersion()`.

Reply:
What about `getRange()`? or we can folow the record semantics and name it `versionRange()` even though it is a bit unusual at the first place, I found this often convenient to have an interface and use a record as the implementation, then these interface methods are automatically implemented.

Reply:
For me `getRange()` is not more clear than `getVersion()`.
The point about records is true and actually I like using records but since the implementations are not records we don't have an immediate benefit from that and would would just be a single method with a different style.
But I still find this idea interesting and created https://github.com/eclipse-pde/eclipse.pde/pull/1341 to explore it further.

In the meantime my suggestion is to continue with `getVersion()`. Since we have enough time to complete that in this release cycle we can immediately replace that method e.g. by `version()` or `versionRange()` in https://github.com/eclipse-pde/eclipse.pde/pull/1341.